### DISCUSSION TITLE
how to run alpine in qemu with multi-core

### DISCUSSION THREAD

**Discussion by Angel0726** [termux/termux-app#2545] [Q&A] [answered]
I found that alpine in qemu is too slow; what's i use `top` in termux, i find only one core works. so how to run alpine in qemu with multi-core. I test `qemu-system-x86_64 -hda virt-alpine.img -boot c -m 1024 -smp 6,sockets=6`, but it does not work
![image](https://user-images.githubusercontent.com/32816780/150999190-d0a2c218-5bba-471d-8a66-1bd9a1558a64.png)

**Comment by nisheri-ascar** [comment]
`-accel tcg;threads=multi` should do it

**Reply by Angel0726** [reply]
it doesn't work

**Reply by SDRausty** [reply]
Different Android versions have different permission schemes. 

What are your Android versions @Angel0726 and @nisheri-ascar?

**Reply by ghost** [reply]
> it doesn't work
> 

The comment who suggested a solution has typo. try `-accel tcg,thread=multi`

**Comment by sylirre** [comment, marked as answer]
Indeed the solution is `--accel tcg,thread=multi` like said in another comment. However MTTCG could be unstable. I never had it working properly.

Ref: https://wiki.qemu.org/Features/tcg-multithread
